src/components/subscriptions/details/Drawer.js

```diff
@@ -269,12 +244,23 @@ function UsagesDetails(props) {
         <>
             {selectedSubscription &&
                 selectedSubscription.usages.length > 0 &&
-                selectedSubscription.usages.map((item, _) => {
-                    return (
-                        <Typography>
-                            {item.usage} {item.resource_type.unit}
-                        </Typography>
-                    );
+                selectedSubscription.usages.map((item, index) => {
+                    // Only format data usage to human readable format
+                    let resourceInBytes =
+                        item.resource_type.unit.toLowerCase() === "bytes";
+                    if (resourceInBytes) {
+                        return (
+                            <Typography key={index}>
+                                {formatFileSize(item.usage)}
+                            </Typography>
+                        );
+                    } else {
+                        return (
+                            <Typography>
+                                {item.usage} {item.resource_type.unit}
+                            </Typography>
+                        );
+                    }
```

Comment:
Each element returned in this `map` array should have a `key`, including the `Typography` for CPUs display.

So I would nest the `resourceInBytes` check inside the `Typography` tag:
```suggestion
                    return (
                        <Typography key={index}>
                            {resourceInBytes
                                ? formatFileSize(item.usage)
                                : `${item.usage} ${item.resource_type.unit}`}
                        </Typography>
                    );
```

Reply:
Yes this looks much better. I'll implement this before merging.